cogs/strawpoll.py

```diff
@@ -0,0 +1,292 @@
+            self._delete_expired_poll_sessions(current_time)
+            self._remove_done_poll_tasks()
+            await asyncio.sleep(5)
+        for poll in self.poll_session_tasks.keys():
+            self.poll_session_tasks[poll].cancel()
+        
+    async def _check_reacts(self, poll):
+        settings = self.settings[poll.message.server.id]
+        refresh_emoji = settings['refresh_emoji']
+        try:
+            while True:
+                react = await self.bot.wait_for_reaction(refresh_emoji, 
+                    message=poll.message)
+                await poll.update_results()
+                await asyncio.sleep(0.5) # don't remove reaction too fast
+                await self.bot.remove_reaction(
+                    poll.message, refresh_emoji, react.user)
+        except CancelledError:
+            await self.bot.clear_reactions(poll.message)
+            raise CancelledError("Poll cancelled")
+            
+    async def _create_poll(self, message, channel, author, poll_id):
+        settings = self.settings[message.server.id]
+        refresh_emoji = settings['refresh_emoji']
+        poll_length = settings['poll_length']
+        poll = _Poll(self.bot, settings, 
+                message, author, poll_id, poll_length)
+        while time()-poll.start_time < poll_length:
+            await poll.update_results()
+            await asyncio.sleep(1)
+        await self.bot.delete_message(poll.message)
+        poll.message = await self.bot.send_message(channel,
+            embed=discord.Embed(title="Loading results..."))
+        await poll.update_results()
+        await self.bot.add_reaction(poll.message, refresh_emoji)
+        self.poll_sessions.append(poll)
+    
+    @commands.command(pass_context=True, no_pm=True)
+    async def strawpoll(self, ctx, *text):
+        """
+        Host a poll on Strawpoll.me with live results
+
+        Usage: strawpoll [m] title;option 1;option 2 (...)
+        
+        Options:
+            m       Allow multiple options to be selected
+        """
+        if ctx.message.server.id not in self.settings:
+            self._new_server_settings(ctx.message.server.id)
+        multi = False
+        if len(text) <= 0:
+            await self.bot.send_cmd_help(ctx)
+            return
+        if text[0] is 'm':
+            multi = True
+            text = text[1:]
+        poll = ' '.join(text).split(';', 1)
+        if len(poll) != 2:
+            await self.bot.send_cmd_help(ctx)
+            return
+        options = poll[1].split(';')
+        # ~ fancy way of removing empty strings ~
+        # if a string is empty, it evaluates to false.
+        # so, if an option from options has content, then
+        # assign that option for itself, otherwise it wont get
+        # included in the splice. (took me a sec to parse it lol)
+        options[:] = [option for option in options if option.strip()]
+        response = _post_poll(poll[0], options, multi)
```

Comment:
For code clarity, put `title=poll[0]`
The other parameters are self-explanatory.

Reply:
good idea!